### ISSUE TITLE
S1075: Mark Span pattern-matching test cases wrongly marked as FN as "Compliant"

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#6379]
EDIT: This case is not an FN.
The rule does not check this declaration because there is no part of the variable name that matches [the checked variable names set](https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/src/SonarAnalyzer.Common/Rules/UriShouldNotBeHardcodedBase.cs#L44); thus it's a compliant case.

### Description

FN for S1075 (UriShouldNotBeHardcoded).

Using C#11 `is` on `Span` types, the rule is not raising an issue.

### Repro steps

```csharp
    public void SpanMatch(Span<char> span, ReadOnlySpan<char> readonlySpan, string simpleString)
    {
        var a = span is """\\my-network-drive\folder\file.txt"""               // FN
                || readonlySpan is """\\my-network-drive\folder\file.txt"""    // FN
                || simpleString is @"\\my-network-drive/folder/file.txt";      // FN
    }
```

### Expected behavior

The rule should raise an issue.

### Actual behavior

The rule does not raise an issue.
